sylvia/src/multitest.rs

```diff
@@ -1,13 +1,53 @@
 use std::cell::{Ref, RefCell, RefMut};
+use std::fmt::{Debug, Display};
 use std::marker::PhantomData;
 
-use cosmwasm_std::{Addr, BlockInfo, Coin};
-use cw_multi_test::Executor;
+use cosmwasm_std::testing::{MockApi, MockStorage};
+use cosmwasm_std::{
+    Addr, Api, BlockInfo, Coin, CustomQuery, Empty, GovMsg, IbcMsg, IbcQuery, Storage,
+};
+use cw_multi_test::{
+    BankKeeper, DistributionKeeper, Executor, FailingModule, Router, StakeKeeper, WasmKeeper,
+};
+use schemars::JsonSchema;
+use serde::de::DeserializeOwned;
 use serde::Serialize;
 
-#[derive(Default)]
-pub struct App {
-    pub app: RefCell<cw_multi_test::App>,
+/// Type alias for default build `App` to make its storing simpler in typical scenario
+pub type BasicApp<ExecC = Empty, QueryC = Empty> = App<
+    BankKeeper,
+    MockApi,
+    MockStorage,
+    FailingModule<ExecC, QueryC, Empty>,
+    WasmKeeper<ExecC, QueryC>,
+    StakeKeeper,
+    DistributionKeeper,
+    FailingModule<IbcMsg, IbcQuery, Empty>,
+    FailingModule<GovMsg, Empty, Empty>,
+>;
+
+// This exist to silence `clippy::type-complexity`.
+type CWApp<Bank, Api, Storage, Custom, Wasm, Staking, Distr, Ibc, Gov> =
+    RefCell<cw_multi_test::App<Bank, Api, Storage, Custom, Wasm, Staking, Distr, Ibc, Gov>>;
+
+pub struct App<
+    Bank = BankKeeper,
+    Api = MockApi,
+    Storage = MockStorage,
+    Custom = FailingModule<Empty, Empty, Empty>,
+    Wasm = WasmKeeper<Empty, Empty>,
+    Staking = StakeKeeper,
+    Distr = DistributionKeeper,
+    Ibc = FailingModule<IbcMsg, IbcQuery, Empty>,
+    Gov = FailingModule<GovMsg, Empty, Empty>,
+> {
+    pub app: CWApp<Bank, Api, Storage, Custom, Wasm, Staking, Distr, Ibc, Gov>,
```

Comment:
I am pretty sure it is not what you want to keep it. You are getting into general hell.

This should most probably be:

```rust
pub struct App<MtApp> {
  app: RefCell<MtApp>,
}
```

So you don't care about the app's detailed type for most cases. Then what you do is you trait-bout the `MtApp` type depending on context.

And for the sake of whatever you feel fitting... stop creating public fields. Public fields should be only for data transfer types and nothing else. Particularly not for logic-wrapping types.

Rule of thumb - only create public fields if you just want type aggregating some common data which otherwise would be a tuple but you want your fields named. Otherwise always private.